test/include/dlaf_test/matrix/util_tile_setup.h

```diff
@@ -0,0 +1,90 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#include <functional>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include "gtest/gtest.h"
+#include "dlaf/tile.h"
+#include "dlaf/traits.h"
+#include "dlaf_test/matrix/util_tile.h"
+#include "dlaf_test/matrix/util_tile_blas.h"
+
+namespace dlaf {
+namespace matrix {
+namespace test {
+using namespace dlaf;
+
+/// Create a read/write tile and fill with selected values
+///
+/// @pre val argument is an index of type const TileElementIndex&,
+/// @pre val return type should be T;
+/// @pre size is the dimension of the tile to be created (type: TileElementSize);
+/// @pre ld is the leading dimension of the tile to be created.
+template <class T, class ElementGetter>
+Tile<T, Device::CPU> setup_tile(ElementGetter val, const TileElementSize& size, const SizeType& ld) {
+  Tile<T, Device::CPU> support = create_tile<T>(size, ld);
+  set(support, val);
+  return support;
+}
+
+/// Create a read/write tile and fill with selected values
+///
+/// @pre val argument is an index of type const TileElementIndex&,
+/// @pre val return type should be T;
+/// @pre size is the dimension of the tile to be created (type: TileElementSize);
+/// @pre ld is the leading dimension of the tile to be created,
+/// @pre op is the blas::Op to be applied to the tile.
+template <class T, class ElementGetter>
+Tile<T, Device::CPU> setup_tile(ElementGetter val, const TileElementSize& size, const SizeType& ld,
+                                const blas::Op& op) {
+  Tile<T, Device::CPU> support = create_tile<T>(size, ld);
+  set(support, val, op);
+  return support;
+}
+
+/// Create a read-only tile and fill with selected values
+///
+/// @pre val argument is an index of type const TileElementIndex&,
+/// @pre val return type should be T;
+/// @pre size is the dimension of the tile to be created (type: TileElementSize);
+/// @pre ld is the leading dimension of the tile to be created.
+template <class T, class CT = const T, class ElementGetter>
+Tile<CT, Device::CPU> setup_readonly_tile(ElementGetter val, const TileElementSize& size,
```

Comment:
IMHO given what the function is doing, the name is a bit misleading.

By looking at the implementation, this function allocates a new tile of type T and then move this into another tile with the type `CT`, which by default is `const T`. If I call this function with `setup_readonly_tile<double, double>` it won't return a readonly tile.

A starting point for fixing this could be renaming `CT` to make more clear that it is the output type for the tile, and not the CONST TYPE. Then, it may be easier to think of a better name for the function.

Reply:
I agree the name should be setupTile.
@albestro, @Sely85 Regarding the type, is it not better to have just the output type (calling it simply `T`), and use `std::remove_const_t` to get the type to first construct the tile?

Reply:
Yep, I completely agree.